skimage/measure/fit.py

```diff
@@ -664,7 +664,7 @@ def ransac(data, model_class, min_samples, residual_threshold,
 
         where `success` indicates whether the model estimation succeeded
         (`True` or `None` for success, `False` for failure).
-    min_samples : int
+    min_samples : int or float in range [0, 1]
```

Comment:
this is the kind of magic we should avoid. make `min_samples=None`. `min_samples` should mean `min_samples`. 

@jni wanted to make these keyword only agruments. Lets do that! 
Add a new parameter for the new functionality.

Reply:
I'm really sorry for asking so much back and forth, but I feel strongly against this kind of magic.

I'm not against convenience, but for this particular case, it is coming at the expense of readability, and maintainability in the future.

You are correct, it is a one-liner, but it has serious implication for adding new functionality in the future.

Reply:
in such case, the variable should be `nim_num_sample`... as the variable name is now I feel comfortable to set `min_samples` as 20 of 50 or just 0.5~50% of 50

Reply:
I don't understand what `nim` means. Is it shorthand for something?

Reply:
@hmaarrfk @Borda I'm on the fence for this one. Specifically, my definition of magic is that involves guessing. Is any guessing involved here? Could do [0, 1] -> fraction, (1, None) -> int.

But, earlier I argued that someone who meant 1 as the minimum might have used 0.5 because they were unsure whether 1 is included or not when we say `min_samples=1`.

I think my favourite option is something like what @hmaarrfk suggested, but with @Borda's renaming suggestion:

```python
ransac(data, ..., *, min_num_samples=None, min_fraction_samples=None):
```

and deprecate the existing `min_samples` argument. It would be hard to express that one or the other of the above would be a required keyword argument.

I dunno, it's a hard situation. @scikit-image/core comments welcome here! @Borda thanks for your patience with this! Since it's an API issue and, as you can see, changing the API later is hard, we want to make sure we get this right before merging.

Reply:
@hmaarrfk nim is a typo of min, I'm pretty sure. :joy:

Reply:
I really don't mean to discourage this kind of convenience.

It is certainly useful to be able to define parameters in terms of numbers, or fractions. Quite often, one way of thinking is better for a particular problem.

Python allows us to define many different ways of providing parameters to a function. That said, I don't think the `units` of a named parameter should change based on an arbitrary threshold.

FYI: previously, if a fraction was provided, it would fail:
```
TypeError: 'float' object cannot be interpreted as an integer
```

this would also fail for `min_samples = 0.99999999`.

I also think of it in terms of optimization problems, where it might be useful to try and optimize the result, based on some kind of relaxed integer program. With this hybrid interpretation of min_samples, it requires the user to **undo** the guessing.

Reply:
do you think that distinguishing them as a floating-point number (float/double/...) and fix-point number (int) is not enough?

Reply:
I hadn't realised that the existing behaviour is that float raises an error. @Borda can you confirm that this is indeed the case *for all NumPy >= 1.11*? If so, then I'm sort of ok with type-based polymorphism. This is how polymorphism works in C++, for example.

Reply:
do you mean float larger then 1 will raise error< yes, it is included in the tests